src/throttler-storage-redis.service.ts

```diff
@@ -25,21 +22,49 @@ export class ThrottlerStorageRedisService implements ThrottlerStorageRedis, OnMo
     }
   }
 
+  /**
+   * @description: This method look for members (records) in a Redis set, see: https://redis.io/docs/data-types/sets/
+   *
+   * @param key
+   */
   async getRecord(key: string): Promise<number[]> {
-    const ttls = (
-      await this.redis.scan(
-        0,
-        'MATCH',
-        `${this.redis?.options?.keyPrefix}${key}:*`,
-        'COUNT',
-        this.scanCount,
-      )
-    ).pop();
-    return (ttls as string[]).map((k) => parseInt(k.split(':').pop())).sort();
+    const setMembers = await this.redis.smembers(key);
+    const now = +new Date();
```

Comment:
```suggestion
    const now = new Date().getTime();
```

Reply:
`Date.now()` is better to use here as it returns a timestamp as a number.